strings/base_error.h

```diff
@@ -430,6 +446,18 @@ WINRT_EXPORT namespace winrt
         hresult_canceled(take_ownership_from_abi_t WINRT_IMPL_SOURCE_LOCATION_ARGS) noexcept : hresult_error(impl::error_canceled, take_ownership_from_abi WINRT_IMPL_SOURCE_LOCATION_FORWARD) {}
     };
 
+    struct non_originating_hresult_canceled : hresult_error
```

Comment:
i think making hresult_error a templated class would break a lot of code that already does 
```
try {}
catch(winrt::hresult_error& e)

```

so that's why i didn't go with that approach.